src/main/java/liquibase/ext/mongodb/statement/InsertOneStatement.java

```diff
@@ -20,59 +20,66 @@
  * #L%
  */
 
-import com.mongodb.client.MongoCollection;
-import liquibase.ext.mongodb.database.MongoConnection;
-import liquibase.nosql.statement.NoSqlExecuteStatement;
+import com.mongodb.MongoException;
+import com.mongodb.MongoWriteException;
+import com.mongodb.WriteError;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import org.bson.BsonDocument;
 import org.bson.Document;
 
-import static java.util.Optional.ofNullable;
+import java.util.List;
+
+import static java.util.Objects.nonNull;
+import static java.util.Collections.singletonList;
 import static liquibase.ext.mongodb.statement.BsonUtils.orEmptyDocument;
 
+/**
+ * Inserts a document via the database runCommand method
+ * For a list of supported options see the reference page:
+ * https://docs.mongodb.com/manual/reference/command/insert/
+ */
 @Getter
 @EqualsAndHashCode(callSuper = true)
-public class InsertOneStatement extends AbstractCollectionStatement
-implements NoSqlExecuteStatement<MongoConnection> {
+public class InsertOneStatement extends AbstractRunCommandStatement {
 
-    public static final String COMMAND_NAME = "insertOne";
-
-    private final Document document;
-    private final Document options;
+    public static final String COMMAND_NAME = "insert";
 
     public InsertOneStatement(final String collectionName, final String document, final String options) {
         this(collectionName, orEmptyDocument(document), orEmptyDocument(options));
     }
 
     public InsertOneStatement(final String collectionName, final Document document, final Document options) {
-        super(collectionName);
-        this.document = document;
-        this.options = options;
+        super(BsonUtils.toCommand(COMMAND_NAME, collectionName, combine(document, options)));
```

Comment:
I'm combining the `documents` and other options into a single `Document` which `BsonUtils` then makes the command from. I think we need to be careful not to add a proliferation of methods there for logic that can be done in the statements themselves.
Also - the "abstract base-class pattern" here is forcing a certain style that I'm not 100% comfortable with.